File: Javascript/7-Kyu/Sort-the-vowels!.js
// Sort the vowels!
function sortVowels(s){
    
    let vowels = 'aeiouAEIOU';
    let sorted = '';

    if ( typeof s === 'string' ) {
        for ( let i = 0; i < s.length; i++ ) {
            if ( vowels.includes(s[i]) ) {
                sorted = sorted.concat(`|${s[i]}\n`);
            } else {
                sorted = sorted.concat(`${s[i]}|\n`);
            }

        };
    };

    return sorted.trim('\n');

};

// Tests
console.log(sortVowels('Codewars'));  // --> 'C|\n|o\nd|\n|e\nw|\n|a\nr|\ns|'
console.log(sortVowels('Rnd Te5T')); // --> 'R|\nn|\nd|\n |\nT|\n|e\n5|\nT|'
console.log(sortVowels(1337)); // --> ''
console.log(sortVowels(undefined)); // --> ''

File: Python/7-Kyu/Sort-the-vowels!.py
# Sort the Vowels!
def sort_vowels(s):
    
    sortedStr = ''

    if type(s) == str:
        for i in s:
            sortedStr += f'\n|{i}' if i in 'aeiouAEIOU' else f'\n{i}|'

    return sortedStr.strip('\n')

# Tests
print(sort_vowels('Codewars'))  # --> 'C|\n|o\nd|\n|e\nw|\n|a\nr|\ns|'
print(sort_vowels('Rnd Te5T'))  # --> 'R|\nn|\nd|\n |\nT|\n|e\n5|\nT|'
print(sort_vowels(1337));  # --> ''
print(sort_vowels(None));  # --> ''
